ARCHITECTURE.md

```diff
@@ -0,0 +1,352 @@
+# smol-rs Architecture
+
+The architecture of [`smol-rs`].
+
+This document describes the architecture of [`smol-rs`] and its crates on a high
```

Comment:
we follow 100 chars limit for lines in code so I think we should follow the same in markdown files. Short lines make the document look longer and people can be put off by size of the reading. :)

Reply:
Generally I follow a limit for 80 chars in Markdown files, as some terminals rely on this.

Reply:
These days? Which terminals? 😯

In any case, they'll have the same issues with the code. It's best to be consistent.